airbyte-integrations/connectors/source-sentry/unit_tests/integration/test_events.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+# Import conftest helper
+import sys
+from datetime import datetime, timezone
+from pathlib import Path
+from unittest import TestCase
+
+import freezegun
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.test.catalog_builder import CatalogBuilder
+from airbyte_cdk.test.entrypoint_wrapper import read
+from airbyte_cdk.test.mock_http import HttpMocker
+
+# Import builders
+from integration.config import ConfigBuilder
+from integration.request_builder import SentryRequestBuilder
+from integration.response_builder import create_response, error_response
+
+
+sys.path.insert(0, str(Path(__file__).parent.parent))
+from conftest import get_source
+
+
+# Test constants
+_NOW = datetime.now(timezone.utc)
+_STREAM_NAME = "events"
+_ORGANIZATION = "test-org"
+_PROJECT = "test-project"
+_AUTH_TOKEN = "test_token_abc123"
+
+
+@freezegun.freeze_time(_NOW.isoformat())
+class TestEventsStream(TestCase):
```

Comment:
The `events` stream is an incremental sync that allows for passing the date range as query parameters in the API request.

We currently have a test case `test_full_refresh_single_page` which validates the first sync attempt in which we have no state message. But we should also be modifying that test to validate that `start` and `end` query parameters are included in the outbound request. We should also validate the state message we receive back

We also should add a second test where we pass in state with a later date  and validate that the later date is used in the outbound API request. We should also validate the state message we receive back